pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -1660,10 +1640,25 @@ static String getActualColumnName(String rawTableName, String columnName, @Nulla
       return columnName;
     }
     String columnNameToCheck;
-    if (columnName.regionMatches(ignoreCase, 0, rawTableName, 0, rawTableName.length())
-        && columnName.length() > rawTableName.length() && columnName.charAt(rawTableName.length()) == '.') {
-      columnNameToCheck = ignoreCase ? columnName.substring(rawTableName.length() + 1).toLowerCase()
-          : columnName.substring(rawTableName.length() + 1);
+    String resolvedColumnName = columnName;
+    if (rawTableName.contains(".")) { // table name has database prefix
+      String databaseName = rawTableName.split("\\.")[0];
```

Comment:
What if database, table or column name has '.' in its name? or is it allowed to have '.'. I think currently there is no sanity check to block this. The previous logic above used the name.length() to split.

Reply:
Previous logic is still valid and present at new line # 1657. 
This code is only trying to handle the case when user queries something like `select tb1.col1 from db1.tb1` or `select tb1.col1 from tb1` with right database header. In such cases we need to prefix the whole column literal with the database name as the following logic is based on rawTableName which is the physical table name and will have database prefix.

Reply:
As for the sanity check, the actual database name or logical table name will not have '.' as that's ensured during table creation by `TableConfigUtils.validateTableName`.
But I get your point for the column name with '.' case. Will update the condition logic for `if column name only has table prefix` case.
Nice catch!